src/main/java/io/vertx/core/parsetools/RecordTooLongException.java

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright (c) 2011-2018 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ */
+package io.vertx.core.parsetools;
+public class RecordTooLongException extends RuntimeException {
```

Comment:
can you just use a `VertxException` instead ?

Reply:
or better an `IllegalStateException`

Reply:
Sure